src/main/native/cpp/analysis/FilteringUtils.cpp

```diff
@@ -91,3 +138,168 @@ units::second_t sysid::GetMeanTimeDelta(const Storage& data) {
 
   return std::accumulate(dts.begin(), dts.end(), 0_s) / dts.size();
 }
+
+void sysid::TrimQuasistaticData(std::vector<PreparedData>* data,
+                                double motionThreshold) {
+  data->erase(std::remove_if(data->begin(), data->end(),
+                             [motionThreshold](const auto& pt) {
+                               return std::abs(pt.voltage) <= 0 ||
+                                      std::abs(pt.velocity) < motionThreshold;
+                             }),
+              data->end());
+}
+
+void sysid::ApplyMedianFilter(std::vector<PreparedData>* data, int window) {
+  size_t step = window / 2;
+  frc::MedianFilter<double> medianFilter(window);
+
+  // Load the median filter with the first value, step times for accurate
+  // behaviour at the start.
+
+  for (int i = 0; i < step; i++) {
+    medianFilter.Calculate(data->at(i).velocity);
```

Comment:
Should this use `at(0)`, not `at(i)`? Only the first value should be replicated. Otherwise, you end up going backward since the next for loop starts at 0.